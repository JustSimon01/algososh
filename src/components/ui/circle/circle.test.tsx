import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Тестирование компонента Circle", () => {
  test("Отрисовка элемента без буквы", () => {
    const circle = renderer.create(<Circle />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента с буквами", () => {
    const circle = renderer.create(<Circle letter="A" />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента с head", () => {
    const circle = renderer.create(<Circle head="A" />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента с react-элементом в head", () => {
    const circle = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента с tail", () => {
    const circle = renderer.create(<Circle tail="A" />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента с react-элементом в tail", () => {
    const circle = renderer.create(<Circle tail={<Circle />} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента с index", () => {
    const circle = renderer.create(<Circle index={1} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента с пропом 'isSmall ===  true'", () => {
    const circle = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента в состоянии default", () => {
    const circle = renderer.create(<Circle state={ElementStates.Default} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента в состоянии changing", () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  test("Отрисовка элемента в состоянии modified", () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
    expect(circle).toMatchSnapshot();
  });
});
