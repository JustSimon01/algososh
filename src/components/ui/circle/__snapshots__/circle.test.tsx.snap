// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle Отрисовка элемента без буквы 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle   default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  />
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента в состоянии changing 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle   changing"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  />
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента в состоянии default 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle   default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  />
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента в состоянии modified 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle   modified"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  />
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента с head 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="testedCircleHead"
  >
    A
  </div>
  <div
    className="circle   default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  />
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента с index 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle   default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="testedCircleTail"
  />
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента с react-элементом в head 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  >
    <div
      className="content "
      data-testid="testedCircle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="testedCircleHead"
      />
      <div
        className="circle   default"
        data-testid="testedCircleColor"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="testedCircleTail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  />
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента с react-элементом в tail 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle   default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  >
    <div
      className="content "
      data-testid="testedCircle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="testedCircleHead"
      />
      <div
        className="circle   default"
        data-testid="testedCircleColor"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="testedCircleTail"
      />
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента с tail 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle   default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="testedCircleTail"
  >
    A
  </div>
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента с буквами 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle   default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  />
</div>
`;

exports[`Тестирование компонента Circle Отрисовка элемента с пропом 'isSmall ===  true' 1`] = `
<div
  className="content "
  data-testid="testedCircle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="testedCircleHead"
  />
  <div
    className="circle  small default"
    data-testid="testedCircleColor"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="testedCircleTail"
  />
</div>
`;
